### PR TITLE
Remove llvm 14 support

### LINKED ISSUE #2868: Drop LLVM 14 support

### PR DESCRIPTION
**Pull request by nurmukhametov**
This PR removes LLVM 14 related CI jobs and conditional code across the project.

This also updates LLVM version inside the codespace devcontainer.

This fixes #2868

### CODE REVIEW

**Review by dbabokin — APPROVED** [review]
Looks good.

Though I'm wondering why we are not checking the LLVM version in `CMakeLists.txt` and not reporting if the version in unsupported. Though it will trigger during the build anyway.

File: .github/workflows/ispc-ci.yml [hunk 1]
```diff
@@ -82,14 +82,6 @@ jobs:
       fail-fast: false
       matrix:
         llvm:
-          - version: "14.0"
-            full_version: "14.0.6"
-            short_version: 14
-            opaque_ptr_mode: "OFF"
-          - version: "15.0"
```
[the remark this answers is not in the document]
Comment (nurmukhametov) [reply .github/workflows/ispc-ci.yml:89]: It wasn't intentional. Fixed.

**Comment by nurmukhametov** [comment]
> Looks good.
> 
> Though I'm wondering why we are not checking the LLVM version in `CMakeLists.txt` and not reporting if the version in unsupported. Though it will trigger during the build anyway.

I have added this check. BTW, I think we might need to remove all LLVM patches that relate to unsupported LLVM versions.

**Comment by dbabokin** [comment]
> > Looks good.
> > Though I'm wondering why we are not checking the LLVM version in `CMakeLists.txt` and not reporting if the version in unsupported. Though it will trigger during the build anyway.
> 
> I have added this check. BTW, I think we might need to remove all LLVM patches that relate to unsupported LLVM versions.

I don't have strong opinion on that. We need them for the history, but they are in git history anyway.

**Pull request merged**